sklearn/utils/stats.py

```diff
@@ -37,72 +38,86 @@ def _weighted_percentile(array, sample_weight, percentile_rank=50):
         The probability level of the percentile to compute, in percent. Must be between
         0 and 100.
 
+    xp : array_namespace, default=None
+        The standard-compatible namespace for `array`. Default: infer.
+
     Returns
     -------
-    percentile : int if `array` 1D, ndarray if `array` 2D
+    percentile : scalar or 0D array if `array` 1D (or 0D), array if `array` 2D
         Weighted percentile at the requested probability level.
     """
+    xp, _, device = get_namespace_and_device(array)
+    # `sample_weight` should follow `array` for dtypes
+    floating_dtype = _find_matching_floating_dtype(array, xp=xp)
+    array = xp.asarray(array, dtype=floating_dtype, device=device)
```

Comment:
Isn't this going to be a no-op because we got `floating_dtype` and `device` from `array`?

Reply:
Ah yes! I hadn't thought about the int to float case!